framework/src/iotbus/iotbus_uart.c

```diff
@@ -395,7 +418,191 @@ int iotbus_uart_write(iotbus_uart_context_h hnd, const char *buf, unsigned int l
 	return ret;
 }
 
-#ifdef CONFIG_IOTDEV
+static int _uart_sig[IOTBUS_UART_INTR_MAX] = {
+	SIG_IOTBUS_UART_TX_EMPTY,
+	SIG_IOTBUS_UART_TX_RDY,
+	SIG_IOTBUS_UART_RX_AVAIL,
+	SIG_IOTBUS_UART_RECEIVED
+};
+
+static void signal_handler(int sig, siginfo_t *info, void *ctx)
+{
+	void *ptr;
+	iotbus_uart_context_h dev;
+	struct _iotbus_uart_s *handle;
+	int type;
```

Comment:
I think you'd better initialize the type because sometimes it cannot be set from switch conditions.